sanic/request.py

```diff
@@ -164,6 +166,20 @@ def __repr__(self):
     def generate_id(*_):
         return uuid.uuid4()
 
+    def reset_response(self):
+        try:
+            if (
+                self.stream is not None
+                and self.stream.stage is not Stage.HANDLER
+            ):
+                raise ResponseException(
+                    "Cannot reset response because previous response was sent."
+                )
+            self.stream.response = None
```

Comment:
Should also clear the link from response to stream, i.e. `self.stream.response.stream = None`. Always disconnect them both ways to avoid misuse and to allow faster freeing of memory.

Reply:
Fixed, thanks!